### PR TITLE
86cugqgpt/UI/action bar filters

### PR DESCRIPTION
**Pull request by sania-salim**
**Description:**
This PR contains filters for action bar used to search for buses. Filters are conditionally rendered to be visible only on trip listing page. Multilingual support for action bar and filters has also been integrated in here.

**Changes:**
1. Filter buttons, corresponding menus with radio selections and chips
2. Sort button, its menu and badge.
3. Multilingual support integrated with action bar and filters.

### CODE REVIEW

File: src/components/actionBar/filterSort/FilterSort.tsx [hunk 1]
```diff
@@ -1,3 +1,261 @@
...
+    AcUnit,
+    Air,
+    AirlineSeatReclineNormal,
+    Hotel,
+} from '@mui/icons-material';
+
 export default function FilterSort() {
-    return <></>;
+    // anchor state
+    const [anchorElBus, setAnchorElBus] = useState<null | HTMLElement>(null);
```
Comment (RJR-003) [inline src/components/actionBar/filterSort/FilterSort.tsx]: Could change the name anchorElBus to something more readable.

File: src/components/actionBar/filterSort/FilterSort.tsx [hunk 2]
```diff
@@ -1,3 +1,261 @@
...
+                    MenuListProps={{ 'aria-labelledby': 'bus-type-button' }}
+                    onClose={handleMenuClose}
+                >
+                    {/* <FormControl> */}
+                    <RadioGroup
+                        onChange={busFilterHandler}
+                        value={busTypeFilter}
+                    >
+                        <StyledFormControlLabel
+                            label="AC"
```
Comment (RJR-003) [inline src/components/actionBar/filterSort/FilterSort.tsx]: Maybe you can move these labels to constants file.

File: src/components/actionBar/filterSort/FilterSort.tsx [hunk 3]
```diff
@@ -1,3 +1,261 @@
...
+                            control={<Radio value="AC" icon={<AcUnit />} />}
+                        />
+                        <StyledFormControlLabel
+                            label="Non-AC"
+                            control={<Radio value="Non-AC" icon={<Air />} />}
+                        />
+                    </RadioGroup>
+                    {/* </FormControl> */}
```
Comment (RJR-003) [inline src/components/actionBar/filterSort/FilterSort.tsx]: remove unwanted comments

**Review by RJR-003 — CHANGES_REQUESTED** [review]
-Change the icon on the "From" label of the action bar to a more suitable one.
-Change the position of the filter chips because currently the buttons are jumping towards right during filter selection.
-The language in certain places are not changing while change language is given.
-Change the increment and decrement icon on the passenger count to a more suitable one.

File: src/components/actionBar/filterSort/FilterSort.tsx [hunk 4]
```diff
@@ -1,3 +1,261 @@
...
+                <Menu
+                    id="sort-menu"
+                    anchorEl={anchorElSort}
+                    open={openSort}
+                    MenuListProps={{ 'aria-labelledby': 'sort-button' }}
+                    onClose={handleMenuClose}
+                >
+                    <RadioGroup onChange={sortHandler} value={sortBy}>
+                        <StyledFormControlLabel
+                            label="Start date"
```
Comment (nijin-26) [inline src/components/actionBar/filterSort/FilterSort.tsx]: All of these labels could be updated with translations.

**Review by nijin-26 — CHANGES_REQUESTED** [review]
- Change icon for `From` input.
- Give some more right padding for date input.
- The icon between the 'From' & 'To' seems like oval shape when hovered, it would be better if it's circle.
- Not all labels are translated.
- Passengers count input is accepting negative values. 
- I think you could give the functionality to switch the values of `To` & `From` on the icon click.

File: src/components/actionBar/filterSort/FilterSort.tsx (hunk 4 above)
Comment (sania-salim) [reply src/components/actionBar/filterSort/FilterSort.tsx]: All labels have been replaced using usetraslation.

File: src/components/actionBar/filterSort/FilterSort.tsx (hunk 2 above)
Comment (sania-salim) [reply src/components/actionBar/filterSort/FilterSort.tsx]: All labels have been replaced using usetraslation.

File: src/components/actionBar/filterSort/FilterSort.tsx (hunk 4 above)
Comment (sania-salim) [reply src/components/actionBar/filterSort/FilterSort.tsx]: Toggle functionality and Passenger count validation using regex will be covered during functionality phase.

**Review by RJR-003 — APPROVED** [review]

**Review by nijin-26 — APPROVED** [review]

**Pull request merged**